lib/Conversion/TorchToTcp/Elementwise.cpp

```diff
@@ -60,54 +60,144 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
+
+    if (isa<AtenAddScalarOp>(op) || isa<AtenSubScalarOp>(op)) {
+      RankedTensorType tensorResultType =
+          RankedTensorType::get({}, adaptor.getOther().getType());
+      rhs = torch_to_tcp::scalarToTcpTensor(rewriter, op, tensorResultType,
+                                            op.getOther());
+      if (adaptor.getOther().getType().template isa<mlir::FloatType>())
+        // FP rhs is treated as fp64
+        rhs = torch_to_tcp::castTensorToDtype(rewriter, rewriter.getF64Type(),
+                                              outputType, rhs,
+                                              resultType.getElementType());
+      else if (adaptor.getOther().getType().template isa<mlir::IntegerType>())
+        // INT rhs is treated as si64
+        rhs = torch_to_tcp::castTensorToDtype(
+            rewriter, rewriter.getIntegerType(64, true), outputType, rhs,
+            resultType.getElementType());
+      else
+        return rewriter.notifyMatchFailure(op, "Unsupported rhs data type");
+      rhs = torch_to_tcp::broadcastToMatchShapeAndType(rewriter, rhs, lhs);
+    } else {
+      auto inputBType = op.getOther()
+                            .getType()
+                            .template dyn_cast<torch::Torch::ValueTensorType>()
+                            .getDtype();
+      rhs = torch_to_tcp::castTensorToDtype(rewriter, inputBType, outputType,
+                                            rhs, resultType.getElementType());
+      rhs = torch_to_tcp::broadcastToMatchShapeAndType(rewriter, rhs, lhs);
+    }
 
-    if (!skipMultiplyAlpha(op.getAlpha()))
-      return rewriter.notifyMatchFailure(
-          op, "torch ops with alpha != 1 is not yet supported in "
-              "Torch to TCP conversion");
+    lhs = torch_to_tcp::castTensorToDtype(rewriter, inputAType, outputType, lhs,
+                                          resultType.getElementType());
+    lhs = torch_to_tcp::broadcastToMatchShapeAndType(rewriter, lhs, rhs);
+
+    if (!IsMultiplyAlphaOne(op.getAlpha())) {
+      RankedTensorType tensorResultType =
+          RankedTensorType::get({}, adaptor.getAlpha().getType());
+      Value alpha = torch_to_tcp::scalarToTcpTensor(
+          rewriter, op, tensorResultType, op.getAlpha());
+      if (adaptor.getAlpha().getType().template isa<mlir::FloatType>())
+        // FP alpha is treated as fp64
+        alpha = torch_to_tcp::castTensorToDtype(rewriter, rewriter.getF64Type(),
+                                                outputType, alpha,
+                                                resultType.getElementType());
+      else if (adaptor.getAlpha().getType().template isa<mlir::IntegerType>())
+        // INT alpha is treated as si64
+        alpha = torch_to_tcp::castTensorToDtype(
+            rewriter, rewriter.getIntegerType(64, true), outputType, alpha,
+            resultType.getElementType());
+      else
+        return rewriter.notifyMatchFailure(op, "Unsupported alpha data type");
```

Comment:
This pattern keeps getting repeated several times. Can we make a utility out of it?